flag_test.go

```diff
@@ -449,6 +451,19 @@ func testParseWithUnknownFlags(f *FlagSet, t *testing.T) {
 		"stringo", "ovalue",
 		"boole", "true",
 	}
+	wantUnknowns := []string{
+		"--unknown1", "unknown1Value",
+		"--unknown2=unknown2Value",
+		"-u=unknown3Value",
+		"-p", "unknown4Value",
+		"-q",
+		"--unknown7=unknown7value",
+		"--unknown8=unknown8value",
+		"--unknown6", "",
+		"-u", "-u", "-u", "-u", "-u", "",
```

Comment:
I realize this is brought up in the PR description, but I keep thinking about this line. If `u` is an unknown shorthand, everything following it could be a value, e.g. `-Dfoobar`, and it doesn't seem right to assume they are concatenated shorthands.

Reply:
I strongly disagree that concatenating short flags and values should be supported by anything. It is very unexpected to me that this should work where valid short flags are `t` and `o` and the expected value for `o` is `foobar`

```
$ go run main.go -tofoobar
```

This should fail to work both because it's super confusing and terrible design and because it adds a high level of complexity at the flag parsing layer to support both this feature and unknown short flags.

Reply:
@underrun well, it's a pretty old pattern, so I think you'll have to take that up with history. The fact remains that it's a syntax currently supported, and the preservation of unknown flags should be as naive as possible so that the resulting slice can be manually passed to Parse.

Reply:
i don't see how having unknown flags can work while allowing concatenated short flags and values though. it feels like it's reasonable to disallow this if unknown flags are requred. no?

Reply:
@underrun I mocked up an example for you. It's based on my fork, which already has the spirit of the PR applied:

```go
package main

import (
	"github.com/davecgh/go-spew/spew"
	"github.com/spf13/cobra"
	"github.com/spf13/pflag"
)

var rootCmd = &cobra.Command{
	Use: "example",
	RunE: func(cmd *cobra.Command, args []string) error {
		fs := pflag.NewFlagSet("unknowns", pflag.ContinueOnError)
		fs.BoolS("foo", "f", false, "toggle foo")
		fs.BoolS("bar", "b", false, "toggle bar")
		err := fs.Parse(cmd.Flags().GetUnknownFlags())
		if err != nil {
			return err
		}
		spew.Dump(fs.GetAllFlags())
		return nil
	},
	FParseErrWhitelist: cobra.FParseErrWhitelist(pflag.ParseErrorsWhitelist{UnknownFlags: true}),
}

func main() {
	cobra.CheckErr(rootCmd.Execute())
}
```

```shell
$ go run main.go -fb
```

Reply:
@underrun I'm not sure I fully follow, but you did help me stumble on another bug with this implementation; if the unknown flag has a value, this current implementation will swallow the value and consider it an argument.

Works: `-Dfoobar`
Doesn't work: `-D foobar`, `--stringVal foobar`

Reply:
@underrun I think I better understand your comment now. Essentially this comes down to an inability to surface values that are separated by a space, because they are indistinguishable from arguments, and there is no way to determine if it's associated to an unknown flag type. A best effort is all that can be done, and that is achieved if the user includes `=` when assigning a value to an unknown.